apis/core/lib/domain/table/create.ts

```diff
@@ -46,10 +46,22 @@ export async function createTable({
 
   const columns = [...csvSource.columns]
 
+  const columnsToAdd : CsvColumn[] = []
+
+  resource.out(csvw.column)
+    .forEach(({ term: columnId }) => {
+      const column = columns
+        .find(({ id }) => id.equals(columnId))
+      if (!column) {
+        throw new Error(`Column ${columnId} not found`)
+      }
+      columnsToAdd.push(column)
+    })
+
```

Comment:
If you use `.map` instead of `.forEach`, you can avoid this weird "build an empty list and then populate it" dance.

Something like:
```typescript
  const columnsToAdd: CsvColumn[] = resource
    .out(csvw.column)
    .map(({ term: columnId }) => {
      const column = columns
        .find(({ id }) => id.equals(columnId))
      if (!column) {
        throw new Error(`Column ${columnId} not found`)
      }
      return column
    })
```

Reply:
Nice, why have I not thought of this